Main.java

```diff
@@ -1,237 +1,448 @@
+			String[] queryArray = queryRules.getText().split(",");
+			mainFoodList.filterByNutrients(Arrays.asList(queryArray));
 		});
-		nutriQuery.getChildren().addAll(nutriDrop, compDrop, queryAmount, nutriQueryButton);
-		
+		nutriQuery.getChildren().addAll(queryRules, nutriQueryButton);
+
 		queries.getChildren().addAll(nameQuery, orText, nutriQuery);
-		
+
 		GridPane.setConstraints(queries, 0, 3);
-		
+
 		Button clearQuery = new Button("Clear Queries");
-		clearQuery.setAlignment(Pos.BASELINE_CENTER);
+		clearQuery.setOnAction(e -> {
+			nameText.clear();
+			queryRules.clear();
+		});
 		GridPane.setConstraints(clearQuery, 1, 3);
-		
+
 		// ANALYSIS
 		Button analyze = new Button("Analyze Meal");
+		analyze.setOnAction(e -> {
+			analyzeMeal();
+		});
 		GridPane.setConstraints(analyze, 3, 3);
-		
-		TextArea mealAnalysis = new TextArea();
-		GridPane.setConstraints(mealAnalysis, 3, 4);
-		
-		
-		grid.getChildren().addAll(loadFileButton, foodItemInsert, foodListTable, 
-				menuListTable, transferButton, queries, clearQuery, analyze, mealAnalysis);
 
-		Scene mainScene = new Scene(grid);		
+		//		TextArea mealAnalysis = new TextArea();
+		//		GridPane.setConstraints(mealAnalysis, 3, 4);
+
+
+		grid.getChildren().addAll(listMenu, foodItemInsert, foodListTable,
+				menuListTable, transferButton, queries, clearQuery, analyze);
+
+		mainScene = new Scene(grid);
 		mainStage.setScene(mainScene);
 		mainStage.show();
 
 	}
 
+	private void loadFile(){
+		FileChooser fc = new FileChooser();
+		fc.getExtensionFilters().add(new ExtensionFilter("CSV Files", "*.csv", "*.txt"));
+		fc.setTitle("Load Food List");
+		File file = fc.showOpenDialog(mainStage);
+		if (file != null) {
+			mainFoodList.loadFoodItems(file.toString());
+		}
+
+		else {
+			System.out.println("File not found");
+		}
+	}
+
+	private void saveFile() {
+		FileChooser fc = new FileChooser();
+		fc.setTitle("Save Food List");
+		File file = fc.showSaveDialog(mainStage);
+		if (file != null) {
+			mainFoodList.saveFoodItems(file.toString());
+		}
+		else {
+			System.out.println("File not found");
+		}
+	}
+
+	private void addCustomFood() {
+		FoodItem addedFood = new FoodItem(insertId.getText(), insertName.getText());
+		addedFood.addNutrient("calories", new Double(insertCals.getText()));
+		addedFood.addNutrient("carbohydrate", new Double(insertCarbs.getText()));
+		addedFood.addNutrient("fat", new Double(insertFat.getText()));
+		addedFood.addNutrient("protein", new Double(insertProtein.getText()));
+		addedFood.addNutrient("fiber", new Double(insertFiber.getText()));
+		tempFoodList.add(addedFood);
+		foodListTable.refresh();
+		
+	}
+
+	private void clearFoodEntry() {
+		insertId.clear();
+		insertName.clear();
+		insertCals.clear();
+		insertCarbs.clear();
+		insertFat.clear();
+		insertProtein.clear();
+		insertFiber.clear();
+	}
+
+	private void nameQuery() {
+		tempFoodList = FXCollections.observableArrayList(mainFoodList.filterByName(nameText.getText()));

```

Comment:
Won't you want to call the filterByName() method that exists on the FoodData object in order to get the filtered list?

Reply:
filterByName method is used in line 419